File: SDK/NodeSDK/flowpay_sdk.js
const axios = require('axios');

class FlowPayClient {
    /**
     * A simple Node.js SDK client for interacting with the FlowPay API.
     *
     * @example
     * const main = async () => {
     * try {
     * const client = new FlowPayClient({ apiKey: "sk_your_api_key_here" });
     * const response = await client.makePayment(100.50);
     * console.log("Payment successful:", response);
     * } catch (error) {
     * console.error(`Payment failed: ${error.message}`);
     * }
     * };
     * main();
     */
    constructor({ apiKey, baseUrl = "http://localhost:8080" }) {
        /**
         * Initializes the FlowPay client.
         *
         * @param {string} apiKey - The merchant's secret API key (prefixed with 'sk_').
         * @param {string} [baseUrl="http://localhost:8080"] - The base URL of the FlowPay API.
         */
        if (!apiKey) {
            throw new Error("API key is required.");
        }

        // Create an axios instance with pre-configured settings
        this.api = axios.create({
            baseURL: baseUrl,
            headers: {
                "Content-Type": "application/json",
                "X-API-Key": apiKey,
            },
        });
    }

    /**
     * Creates a new payment.
     *
     * @param {number} amount - The amount for the payment. Must be positive.
     * @returns {Promise<object>} A promise that resolves to an object containing the
     * payment response from the API.
     * @throws {Error} If the payment amount is not positive, or if an API/network error occurs.
     */
    async makePayment(amount) {
        if (amount <= 0) {
            throw new Error("Payment amount must be a positive number.");
        }

        const payload = { paymentAmount: amount };

        try {
            const response = await this.api.post('/payments', payload);
            return response.data;
        } catch (error) {
            // Handle axios-specific errors to provide better feedback
            if (error.response) {
                // The request was made and the server responded with a status code
                // that falls out of the range of 2xx
                const errorDetails = error.response.data || error.response.statusText;
                throw new Error(`API Error (Status ${error.response.status}): ${errorDetails}`);
            } else if (error.request) {
                // The request was made but no response was received
                throw new Error("A network error occurred: The server did not respond.");
            } else {
                // Something happened in setting up the request that triggered an Error
                throw new Error(`An unexpected error occurred: ${error.message}`);
            }
        }
    }
}

module.exports = { FlowPayClient };

File: SDK/PythonSDK/flowpay_sdk.py
import requests
from typing import Dict, Any

class FlowPayClient:
    """
    A simple Python SDK client for interacting with the FlowPay API.

    Usage:
        client = FlowPayClient(api_key="sk_your_api_key_here")
        try:
            response = client.make_payment(amount=100.50)
            print("Payment successful:", response)
        except Exception as e:
            print(f"Payment failed: {e}")
    """

    def __init__(self, api_key: str, base_url: str = "https://flowpay-production.up.railway.app"):
        """
        Initializes the FlowPay client.

        Args:
            api_key (str): The merchant's secret API key (prefixed with 'sk_').
            base_url (str, optional): The base URL of the FlowPay API.
                                      Defaults to "http://localhost:8080".
        
        Raises:
            ValueError: If the API key is not provided.
        """
        if not api_key:
            raise ValueError("API key is required.")

        self.base_url = base_url
        self._session = requests.Session()
        self._session.headers.update({
            "Content-Type": "application/json",
            "X-API-Key": api_key
        })

    def make_payment(self, amount: float) -> Dict[str, Any]:
        """
        Creates a new payment.

        Args:
            amount (float): The amount for the payment. Must be positive.

        Returns:
            Dict[str, Any]: A dictionary containing the payment response from the API,
                            including transaction_id, status, and new_balance.

        Raises:
            ValueError: If the payment amount is not positive.
            requests.exceptions.RequestException: For network-related errors.
            Exception: For API-level errors (e.g., invalid key, insufficient funds).
        """
        if amount <= 0:
            raise ValueError("Payment amount must be a positive number.")

        payment_url = f"{self.base_url}/payments"
        payload = {"paymentAmount": amount}

        try:
            response = self._session.post(payment_url, json=payload)
            
            response.raise_for_status()

            return response.json()

        except requests.exceptions.HTTPError as http_err:
            error_message = "An unknown error occurred."
            try:
                error_details = response.text
                error_message = f"API Error (Status {response.status_code}): {error_details}"
            except Exception:
                pass
            raise Exception(error_message) from http_err
        
        except requests.exceptions.RequestException as req_err:
            raise Exception(f"A network error occurred: {req_err}") from req_err